### PR TITLE
Drop experimental versioned attributes support

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , can you review this PR? A bit of cleanup before rc1 ...

File: lib/pengine/rules.c [hunk 1]
```diff
@@ -130,11 +130,6 @@ find_expression_type(xmlNode * expr)
 
     } else if (pcmk__str_eq(attr, CRM_ATTR_ROLE, pcmk__str_none)) {
         return role_expr;
-
-#if ENABLE_VERSIONED_ATTRS
-    } else if (pcmk__str_eq(attr, CRM_ATTR_RA_VERSION, pcmk__str_none)) {
-        return version_expr;
```
Comment (nrwahl2) [inline lib/pengine/rules.c:136]: We can add a deprecation note to `expression_type:version_expr`

**Review by nrwahl2 — COMMENTED** [review]
It will be so nice to be rid of all this

File: cts/cts-scheduler.in [hunk 2]
```diff
@@ -380,10 +380,6 @@ TESTS = [
         [ "probe-target-of-failed-migrate_to-1", "Failed migrate_to, target rejoins" ],
         [ "probe-target-of-failed-migrate_to-2", "Failed migrate_to, target rejoined and probed" ],
         [ "partial-live-migration-multiple-active", "Prevent running on multiple nodes due to partial live migration" ],
-
-        # @TODO: If pacemaker implements versioned attributes, uncomment this test
-        #[ "migrate-versioned", "Disable migration for versioned resources" ],
-
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply cts/cts-scheduler.in:386]: good catch, thanks

File: daemons/controld/controld_execd.c [hunk 3]
```diff
@@ -734,10 +734,6 @@ build_operation_update(xmlNode * parent, const lrmd_rsc_info_t *rsc,
         return TRUE;
     }
 
-#if ENABLE_VERSIONED_ATTRS
-    crm_xml_add(xml_op, XML_ATTR_RA_VERSION, metadata->ra_version);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply daemons/controld/controld_execd.c:738]: I forgot about those ...

File: daemons/controld/controld_metadata.c [hunk 4]
```diff
@@ -74,72 +70,10 @@ metadata_cache_reset(GHashTable *mdc)
...
-         */
-        CRM_CHECK(version_format_regex != NULL, return TRUE);
-    }
-
-    return regexec(version_format_regex, version, 0, NULL, 0) == 0;
-}
-#endif
-
 void
 metadata_cache_fini(void)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply daemons/controld/controld_metadata.c:74]: true

File: daemons/controld/controld_metadata.c [hunk 5]
```diff
@@ -225,10 +159,6 @@ controld_cache_metadata(GHashTable *mdc, const lrmd_rsc_info_t *rsc,
         goto err;
     }
 
-#if ENABLE_VERSIONED_ATTRS
-    md->ra_version = ra_version_from_xml(metadata, rsc);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply daemons/controld/controld_metadata.c:229]: yep

File: lib/pengine/rules.c (hunk 1 above)
Comment (kgaillot) [reply lib/pengine/rules.c:136]: good catch

**Comment by kgaillot** [comment]
Added commits on top to address review

**Review by nrwahl2 — APPROVED** [review]
I think that's everything :)

**Pull request merged**